include/osquery/events.h

```diff
@@ -745,18 +752,20 @@ class EventSubscriber : public EventSubscriberPlugin {
    * @param sc The subscription context.
    */
   template <class T, typename C>
-  void subscribe(Status (T::*entry)(const std::shared_ptr<C>&),
-                 const SubscriptionContextRef& sc) {
+  void subscribe(Status (T::*entry)(const std::shared_ptr<C>&, const void*),
```

Comment:
Alternatively, you can overload this, and `addSubscription`; then initialize the `extra_data` field in `Subscription` to a `nullptr` and not have to make all the change where extra_data is not used. It would be fantastic to have the bind support both incantations and only optionally define you callback method to have `extra_data` if you want to use it. 

But we can spend more time on the events API later.